waku/v2/node/wakunode2.nim

```diff
@@ -495,44 +496,34 @@ proc mountRelay*(node: WakuNode,
   
   info "mounting relay", rlnRelayEnabled=rlnRelayEnabled, relayMessages=relayMessages
 
+  ## The default relay topics is the union of
+  ## all configured topics plus the hard-coded defaultTopic(s)
+  wakuRelay.defaultTopics = concat(@[defaultTopic], topics)
+  wakuRelay.rlnRelayEnabled = rlnRelayEnabled
+
   node.switch.mount(wakuRelay, protocolMatcher(WakuRelayCodec))
 
-  if not relayMessages:
+  if relayMessages:
     ## Some nodes may choose not to have the capability to relay messages (e.g. "light" nodes).
     ## All nodes, however, currently require WakuRelay, regardless of desired capabilities.
     ## This is to allow protocol stream negotation with relay-capable nodes to succeed.
     ## Here we mount relay on the switch only, but do not proceed to subscribe to any pubsub
     ## topics. We also never start the relay protocol. node.wakuRelay remains nil.
-    ## @TODO: in future, this WakuRelay dependency will be removed completely
-    return
-
-  node.wakuRelay = wakuRelay
-
-  node.subscribe(defaultTopic, none(TopicHandler))
+    ## @TODO: in future, this WakuRelay dependency will be removed completely  
```

Comment:
Not sure I understand this comment

Reply:
Old comment indicating that we have a dependency on `relay` being mounted for any protocol to work. Hopefully can remove this dependency in future with protocol (peer) management.